Hortense phase2 (#317)
hpcleuven/VscDocumentation: source/gent/tier1_hortense.rst

**Hortense** is the 3rd VSC Tier-1 cluster, following *muk* (hosted by HPC-UGent, 2012-2016)
and *BrENIAC* (hosted by HPC-Leuven, 2016-2022).

The Hortense cluster also has an internal name ``dodrio``.

It is available since 2021, is hosted by Ghent University,
and maintained and supported by the HPC-UGent team.

> In 2023, a second phase was added more than doubling the capacity of the system.

Reviewer: ```suggestion
In 2023 a second phase was added, more than doubling the existing capacity of the system.
```

Hardware details
----------------

> Hortense currently consists of following partitions:

Reviewer: ```suggestion
Hortense currently consists of the following partitions:
```

> - ``dodrio/cpu_rome``: main partition:

Reviewer: ```suggestion
- ``dodrio/cpu_rome``: phase 1 main partition:
```

**Hortense** is the 3rd VSC Tier-1 cluster, following *muk* (hosted by HPC-UGent, 2012-2016)
and *BrENIAC* (hosted by HPC-Leuven, 2016-2022).

The Hortense cluster also has an internal name ``dodrio``.

It is available since 2021, is hosted by Ghent University,
and maintained and supported by the HPC-UGent team.

> In 2023 a second phase was added, more than doubling the existing capacity of the system.

Reviewer 2: In May 2023?

Author: previous paragraph is also vague, so i keep it the same

Reviewer 3: May is mentioned later on. This suffices imho

Hardware details
----------------

Hortense consists of the following partitions:

- ``dodrio/cpu_rome``: phase 1 main partition:
   - 342 workernodes, each with:

>        - 2x 64-core AMD Epyc 7H12 CPU 2.6 GHz (128 cores per node)

Reviewer 2: ```suggestion
       - 2x 64-core AMD Epyc 7H12 CPU 2.6 GHz ("Rome" microarchitecture, 128 cores per node)

       - 2x 64-core AMD Epyc 7H12 CPU 2.6 GHz (128 cores per node)
       - 512 GiB RAM (~4GB/core), no swap
       - 480 GB SSD local disk

> - ``dodrio/cpu_milan``: phase2 main partition:

Reviewer: ```suggestion
- ``dodrio/cpu_milan``: phase 2 main partition:
```

   - 384 workernodes, each with:
       - 2x 64-core AMD Epyc 7763 CPU 2.45 GHz (128 cores per node)
       - 256 GiB RAM (~2GB/core), no swap
       - 480 GB SSD local disk
- ``dodrio/gpu_rome_a100_40``: GPU partition:
   - 20 workernodes, each with:
       - 2x 24-core AMD Epyc 7402 CPU 2.8 GHz (48 cores per node)
       - 4x NVIDIA A100-SXM4 (40 GB GPU memory), NVLink3
       - 256 GiB RAM (~5GB/CPU core), no swap
       - 480 GB SSD local disk

> - ``dodrio/gpu_rome_a100_80``: phase2 GPU partition:

Reviewer: ```suggestion
- ``dodrio/gpu_rome_a100_80``: phase 2 GPU partition:
```

       - 2x 64-core AMD Epyc 7H12 CPU 2.6 GHz (128 cores per node)
       - 512 GiB RAM (~4GB/core), no swap
       - 480 GB SSD local disk
- ``dodrio/cpu_milan``: phase 2 main partition:
   - 384 workernodes, each with:

>        - 2x 64-core AMD Epyc 7763 CPU 2.45 GHz (128 cores per node)

Reviewer 2: - 2x 64-core AMD Epyc 7763 CPU 2.45 GHz ("Milan" microarchitecture, 128 cores per node)

To ensure that the paths which are 'engraved' in your own software installations always start with ``/readonly/``,
for example in scripts or binaries that make part of the installation,
you should install the software using the ``dodrio-bind-readonly`` utility. This allows you to "rename" the path to your

> project scratch directory so it starts with ``/readonly/``, while preserving sort-of write access to it (

Reviewer: what's up with the `(` at the end?

with ``/readonly/``.

Or you can start a new shell session in which ``/readonly/$VSC_SCRATCH_PROJECTS_BASE/...`` is

> accessible with sort-of write permissions:

Reviewer 2: I would only add "sort-of" if you explain why you call it "sort-of"... That's already clarified above, no need to here imho.

For questions and problems related to Tier-1 Hortense, please contact the central
support address for Tier-1 compute: `compute@vscentrum.be <mailto:compute@vscentrum.be>`_.

Phase 2
-------

In May 2023 a second phase was installed, adding 48 more nodes to the ``cpu_rome`` partition,
20 extra GPU nodes with double the CPU and GPU memory in the new ``gpu_rome_a100_80`` partition and
384 nodes using the newer AMD Milan CPUs called ``cpu_milan``.

> The Lustre based scratch storage was also doubled in volume and throughput to a total of 6PB.

Reviewer: ```suggestion
The Lustre based scratch storage was also doubled in volume to a total of 6PB while increasing the overal throughput as well.
```

Author: @itkovian the throughput was also doubled, i'll see if i can rephrase this better

Author: ok, nvm, using your suggestion

With the new GPU nodes, a renaming of the gpu node partitions occured. Users can mosyt likely still
still use the same ``gpu_rome_a100`` partition that now includes all GPU nodes (and only select the

> ``gpu_rome_a100_40`` or ``gpu_rome_a100_80`` for specific cases, e.g. when running multinode GPU jobs

Author: @boegel i only now realised that for multinode jobs, this will not be transparent

Reviewer 2: Yeah, for multi-node GPU jobs, they should probably be submitting to a more specific GPU partition.

Or would Slurm only start jobs within the same actual partition anyway?

Reviewer: Yes, it will not pick nodes from separate partitions afaik.

Author: @itkovian can we set something in the partition definition that multinode jobs won't get started?

Reviewer 4: We use `MaxNodes` for that purpose. 

You can also use multiple partitions for a job and let the users pick: either `gpu_rome_a100_40` or `gpu_rome_a100_80` or you specify both (or use this as default for when the partition has not been set for a GPU job). That's what we currently do on Hydra with a job submit plugin.

or explicitly requiring the larger amount of GPU/CPU memory of the ``gpu_rome_a100_80`` nodes).

> During the May 2023 maintenance, the OS and OFED infiniband stacks were updated to resp. RHEL 8.6

Reviewer 2: Maybe this should go into an "Recent updates" section below?

For questions and problems related to Tier-1 Hortense, please contact the central

> support address for Tier-1 compute: `compute@vscentrum.be <mailto:compute@vscentrum.be>`_.

Reviewer 2: `Resources` and `Getting help` parts should remain as last sections?

> Phase 2

Reviewer 2: ```suggestion
Phase 2 (May 2023)
```

Maybe this should be under a separate "History" section, where you also mention something on `Phase 1`?

-------

In May 2023 a second phase was installed, adding 48 more nodes to the ``cpu_rome`` partition,

> 20 extra GPU nodes with double the CPU and GPU memory in the new ``gpu_rome_a100_80`` partition and

Reviewer 2: ```suggestion
20 extra GPU nodes with double the CPU and GPU memory in the new ``gpu_rome_a100_80`` partition, and

384 nodes using the newer AMD Milan CPUs called ``cpu_milan``.

The Lustre based scratch storage was also also doubled in volume to a total of 6PB
while increasing the overal throughput as well.

With the new GPU nodes, a renaming of the gpu node partitions occured. Users can mosyt likely still
still use the same ``gpu_rome_a100`` partition that now includes all GPU nodes (and only select the
``gpu_rome_a100_40`` or ``gpu_rome_a100_80`` for specific cases, e.g. when running multinode GPU jobs
or explicitly requiring the larger amount of GPU/CPU memory of the ``gpu_rome_a100_80`` nodes).

In the startup period, users are encouraged to try out the ``cpu_milan`` partition to compare performance
and overal functioning with the ``cpu_rome`` partitions.
In a later stage, projects will be given access to either the ``cpu_rome`` partitions
or the ``cpu_milan`` partition.

> With both phases active, the cluster crossed the symbolic threshold of 100000 cores.

Reviewer 2: `100000` -> `100,000`
